tests/phpunit/tests/canonical.php

```diff
@@ -275,6 +275,67 @@ public function test_strict_redirect_guess_404_permalink() {
 		$this->assertFalse( redirect_guess_404_permalink() );
 	}
 
+	/**
+	 * Ensure public posts with custom public statuses are guessed.
+	 *
+	 * @ticket 47911
+	 * @dataProvider data_redirect_guess_404_permalink_with_custom_statuses
+	 *
+	 * @covers ::redirect_guess_404_permalink
+	 */
+	public function test_redirect_guess_404_permalink_with_custom_statuses( $status_args, $redirects ) {
+		register_post_status( 'custom', $status_args );
+
+		$post = self::factory()->post->create(
+			array(
+				'post_title'  => 'custom-status-public-guess-404-permalink',
+				'post_status' => 'custom',
+			)
+		);
+
+		$this->go_to( 'custom-status-public-guess-404-permalink' );
+
+		$expected = $redirects ? get_permalink( $post ) : false;
+
+		$this->assertSame( $expected, redirect_guess_404_permalink() );
+	}
+
+	/**
+	 * Data provider for test_redirect_guess_404_permalink_with_custom_statuses().
+	 *
+	 * return array[] {
+	 *    array Arguments used to register custom status
+	 *    bool  Whether the 404 link is expected to redirect
+	 * }
+	 */
+	public function data_redirect_guess_404_permalink_with_custom_statuses() {
+		return array(
+			'public status'                      => array(
+				array( 'public' => true ),
+				true,
```

Comment:
```suggestion
				'status_args' => array( 'public' => true ),
				'redirects'   => true,
```
For those not using `--testdox`, and particularly those not familiar with these tests, this will immediately indicate which value correspond to each parameter.

Reply:
Done in 01472f29fdb51ba5278383f89524609e8a592cc2